dags/app/user_profile/udf.py

```diff
@@ -0,0 +1,318 @@
+                    tgt.age_range = src.age_range,
+                    tgt.gender = src.gender,
+                    tgt.update_time = CURRENT_TIMESTAMP()
+            WHEN NOT MATCHED THEN
+                INSERT (year, email, organization, job_title, country_or_region, age_range, gender, update_time)
+                VALUES (
+                    EXTRACT(YEAR FROM PARSE_DATE('%F', paid_date)),
+                    src.email,
+                    src.organization,
+                    src.job_title,
+                    src.country_or_region,
+                    src.age_range,
+                    src.gender,
+                    CURRENT_TIMESTAMP()
+                )
+            """
+    job = client.query(query)
+    job.result()
+
+
+def get_task_config(task_type: str) -> str:
+    """
+    根據任務類型獲取對應的標頭和要求。
+
+    Args:
+        task_type: 任務類型 ('organization' 或 'job_title')。
+
+    Returns:
+        包含標頭和要求的元組 (header, requirement)。
+
+    Raises:
+        click.BadParameter: 如果任務類型不支援。
+    """
+    if task_type == "organization":
+        return ORG_REQUIREMENT
+    elif task_type == "job_title":
+        return JOB_REQUIREMENT
+    else:
+        raise ValueError("不支援的工作類型。請選擇 'organization' 或 'job_title'。")
+
+
+def check_gemini_api_key() -> str:
+    GEMINI_API_KEY = Variable.get("GOOGLE_GEMINI_KEY")
+    if not GEMINI_API_KEY:
+        logging.error("錯誤：請設置環境變數 GEMINI_API_KEY")
+        raise ValueError("GEMINI_API_KEY 未設定")
+    return GEMINI_API_KEY
+
+
+def read_kktix_ticket_user_profile(task_type: str) -> list[str]:
+    client = bigquery.Client()
+    try:
+        query = f"""
+            SELECT DISTINCT {task_type}
+            FROM `{USER_PROFILE_TABLE}`
+            WHERE {task_type} IS NOT NULL
+            AND update_time > TIMESTAMP_SUB(CURRENT_TIMESTAMP(), INTERVAL 1 DAY)
+        """
+        rows = client.query(query).result()  # 等 query 完成
+        # 將每一 row 的 task_type 放到 list
+        data_list = [row[task_type] for row in rows]
+
+    except Exception as e:
+        logging.error(f"讀取 BigQuery 時發生錯誤: {e}")
+        raise ValueError(f"讀取 BigQuery 時發生錯誤: {e}")
+
+    return data_list
+
+
+def write_result_to_bigquery(df: pd.DataFrame, task_type: str) -> None:
+    client = bigquery.Client()
+    table_id = f"{BIGQUERY_PROJECT}.{BIGQUERY_DATASET}.kktix_ticket_{task_type}_updates"  # 要建立的暫存表
+
+    job_config = bigquery.LoadJobConfig(
+        write_disposition=bigquery.WriteDisposition.WRITE_TRUNCATE,  # 覆蓋舊表，如果表不存在則建立
+        schema=[
+            bigquery.SchemaField(task_type, "STRING"),
+            bigquery.SchemaField("category", "INT64"),
+        ],
+    )
+    job = client.load_table_from_dataframe(df, table_id, job_config=job_config)
+    job.result()
+
+
+def chunk_data(data, batch_size) -> Generator[list[str], None, None]:
+    for i in range(0, len(data), batch_size):
+        yield data[i : i + batch_size]
+
+
+def call_gemini(model: str, max_output_tokens_on_model: int, prompt: str) -> str | None:
```

Comment:
```suggestion
def get_gemini_response(model: str, max_output_tokens_on_model: int, prompt: str) -> str | None:
```

might not be the best naming but a little bit better I think